actionpack/test/dispatch/request_test.rb

```diff
@@ -116,9 +121,15 @@ class RequestIP < BaseRequestTest
   end
 
   test "remote ip spoof detection" do
-    request = stub_request "HTTP_X_FORWARDED_FOR" => "1.1.1.1",
-                           "HTTP_CLIENT_IP"       => "2.2.2.2"
+    # `stub_request` uses `Rack::Lint` and it has different behavior in Rack 2 and Rack 3.
```

Comment:
I opened https://github.com/rack/rack/pull/2101 to fix this in Rack, if it gets merged we should be able to clean this up

Reply:
Awesome, working great with `export RACK=2.2.8`